examples/c/CMakeLists.txt

```diff
@@ -0,0 +1,49 @@
+# SPDX-License-Identifier: GPL-2.0 OR BSD-3-Clause
+
+cmake_minimum_required(VERSION 3.20)
+project(examples)
+
+# Tell cmake where to find BpfProg module
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/../../tools/cmake)
+
+# Build vendored libbpf
+include(ExternalProject)
+ExternalProject_Add(libbpf
+  PREFIX libbpf
+  SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../libbpf/src
+  CONFIGURE_COMMAND ""
+  BUILD_COMMAND make
+    BUILD_STATIC_ONLY=1
+    OBJDIR=${CMAKE_CURRENT_BINARY_DIR}/libbpf/libbpf
+    DESTDIR=${CMAKE_CURRENT_BINARY_DIR}/libbpf
+    INCLUDEDIR=
+    LIBDIR=
+    UAPIDIR=
+    install
+  BUILD_IN_SOURCE TRUE
+  INSTALL_COMMAND ""
+  STEP_TARGETS build
+)
+
+# Set BpfProg input parameters -- note this is usually not necessary unless
+# you're in a highly vendored environment (like libbpf-bootstrap)
+set(BPFTOOL_EXE ${CMAKE_CURRENT_SOURCE_DIR}/../../tools/bpftool)
+set(GENERATED_VMLINUX ${CMAKE_CURRENT_SOURCE_DIR}/../../vmlinux/vmlinux.h)
+set(LIBBPF_INCLUDE_DIRS ${CMAKE_CURRENT_BINARY_DIR}/libbpf)
+set(LIBBPF_LIBRARIES ${CMAKE_CURRENT_BINARY_DIR}/libbpf/libbpf.a)
+find_package(BpfProg REQUIRED)
+
+# Create an executable for each application
+file(GLOB apps *.bpf.c)
+foreach(app ${apps})
+  cmake_path(GET app FILENAME app_filename)
+  cmake_path(GET app_filename STEM app_stem)
+
+  # Build prog skeleton, depend skeleton on libbpf, and static link libbpf's deps
+  bpf_prog(${app_stem} ${app_stem}.bpf.c)
+  add_dependencies(${app_stem}_prog libbpf-build)
+  target_link_options(${app_stem}_prog INTERFACE -lelf -lz)
```

Comment:
is there any way to automatically include -lelf and -lz when linking with libbpf?

Reply:
Yeah, found a way using a (maybe) hacky string match on `.*\.a`. I think it should work